src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreSummaryRequest.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections.Generic;
+using NuGet.Configuration;
+using NuGet.Logging;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.Commands
+{
+    /// <summary>
+    /// Wrapper for RestoreRequest
+    /// </summary>
+    public class RestoreSummaryRequest
```

Comment:
It contains the extra fields needed for RestoreSummary. RestoreCommand which consumes RestoreRequest doesn't need these fields, it has the product of the caller using the settings and applying them for the RestoreRequest, having settings on the RestoreRequest would make it confusing as to where things should and were coming from.